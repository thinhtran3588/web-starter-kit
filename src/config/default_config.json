{
  "dateFormat": "DD/MM/YYYY",
  "signInOptions": ["email", "phone", "facebook", "google"]
}
