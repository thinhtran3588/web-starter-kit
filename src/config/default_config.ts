export default {
  apiEndpoint: 'http://localhost:3001',
  author: 'Thinh Tran',
  copyRightYear: 2019,
  dateFormat: 'DD/MM/YYYY',
  debounceDelay: 300,
  defaultLanguage: 'en',
  languages: [
    {
      code: 'en',
      name: 'English',
    },
    {
      code: 'vi',
      name: 'Tiếng Việt',
    },
  ],
  rowsPerPageOptions: [10, 20, 50],
  signInOptions: ['email', 'phone', 'facebook', 'google'],
  siteName: 'Web Starter Kit',
};
