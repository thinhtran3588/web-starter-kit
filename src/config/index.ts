// import merge from 'lodash/fp/merge';
import defaultConfig from './default_config.json';
import overrideConfig from './override_config.json';

const regexConfig = {
  regex: {
    password: /^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{6,}$/,
    email: /^[a-z][a-z0-9_.]{5,40}@[a-z0-9]{2,}(.[a-z0-9]{2,4}){1,2}$/,
    phone: /^[0-9]{8,}$/,
  },
};

export const config = Object.freeze({
  ...defaultConfig,
  ...overrideConfig,
  ...regexConfig,
});
