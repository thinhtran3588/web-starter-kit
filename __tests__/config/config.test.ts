import { config } from '@app/config';

describe('pages/Index', () => {
  beforeEach(() => {});

  it('renders successfully', async () => {
    expect(config).toMatchSnapshot();
  });
});
