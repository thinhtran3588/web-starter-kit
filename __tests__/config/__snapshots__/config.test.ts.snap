// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/Index renders successfully 1`] = `
Object {
  "apiEndpoint": "http://localhost:3001",
  "author": "Thinh Tran",
  "contactEmail": "contact@webstarterkit.com",
  "copyRightYear": 2019,
  "dateFormat": "DD/MM/YYYY",
  "dateTimeFormat": "DD/MM/YYYY HH:mm:ss",
  "debounceDelay": 300,
  "defaultCountryCode": "+84",
  "defaultWaitToResend": 30,
  "firebaseConfig": Object {
    "apiKey": "should_be_replaced",
    "appId": "should_be_replaced",
    "authDomain": "should_be_replaced",
    "databaseURL": "should_be_replaced",
    "measurementId": "should_be_replaced",
    "messagingSenderId": "should_be_replaced",
    "projectId": "should_be_replaced",
    "storageBucket": "should_be_replaced",
  },
  "i18n": Object {
    "defaultLang": "en",
    "languages": Array [
      Object {
        "code": "en",
        "name": "English",
      },
      Object {
        "code": "vi",
        "name": "Tiếng Việt",
      },
    ],
  },
  "regex": Object {
    "email": /\\^\\[a-z\\]\\[a-z0-9_\\.\\]\\{0,40\\}@\\[a-z0-9\\]\\{2,\\}\\(\\.\\[a-z0-9\\]\\{2,4\\}\\)\\{1,2\\}\\$/,
    "password": /\\^\\(\\?=\\.\\*\\[A-Za-z\\]\\)\\(\\?=\\.\\*\\\\d\\)\\(\\?=\\.\\*\\[@\\$!%\\*#\\?&\\]\\)\\[A-Za-z\\\\d@\\$!%\\*#\\?&\\]\\{6,20\\}\\$/,
    "phone": /\\^\\[0-9\\]\\{8,\\}\\$/,
    "username": /\\^\\[a-z0-9\\]\\{5,15\\}\\$/,
  },
  "rowsPerPageOptions": Array [
    10,
    20,
    50,
  ],
  "signInOptions": Array [
    "email",
    "phone",
    "facebook",
    "google",
  ],
  "siteName": "Web Starter Kit",
  "storageLocation": Object {
    "base": "",
    "public": "/public",
    "static": "/static",
  },
}
`;
